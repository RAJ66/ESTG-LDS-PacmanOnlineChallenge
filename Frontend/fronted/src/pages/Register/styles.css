.container-register {
  width: 100vw;
  height: 100vh;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  background: var(--color-base);
}

.register-container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  max-width: 400px;
  width: 100%;

  background: var(--color-zero);
  border-radius: 6px;
  padding: 30px;
}

.register-container input {
  margin-top: var(--size-small-medium);
  border: 2px solid var(--color-base);
  border-radius: 4px;
  height: 48px;
  padding: 0 var(--size-small-medium);
  font-size: 16px;
  color: var(--color-base);
}

.register-container button {
  margin-top: var(--size-small);
  border: 0px;
  border-radius: 4px;
  height: var(--size-medium);
  width: var(--size-big);
  font-size: 16px;
  font-weight: bold;
  color: var(--color-zero);
  cursor: pointer;
}
