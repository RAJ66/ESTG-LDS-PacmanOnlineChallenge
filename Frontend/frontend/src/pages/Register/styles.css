.container-register {
  width: 100vw;
  height: 100vh;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  background: var(--color-base);
}

.register-container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  max-width: 400px;
  width: 100%;
  background: var(--color-zero);
  border-radius: 6px;
  padding: 30px;
}

.register-container input {
  margin-top: var(--size-small-medium);
  border: 2px solid var(--color-base);
  border-radius: 4px;
  height: 48px;
  padding: 0 var(--size-small-medium);
  font-size: 16px;
  color: var(--color-base);
}

.register-container button {
  margin-top: var(--size-small);
  border: 0px;
  border-radius: 4px;
  height: var(--size-medium);
  width: var(--size-big);
  font-size: 16px;
  font-weight: bold;
  color: var(--color-zero);
  cursor: pointer;
}

.pacmanFood {
  background: var(--color-third);
}

.pacmanMouth {
  background: var(--color-zero);
}

.pacman {
  margin-right: 250px;
  animation-name: run;
  animation-duration: 5s;
  animation-iteration-count: infinite;
}

@keyframes run {
  0% {
    transform: translateX(0px);
    opacity: 0;
  }
  5% {
    transform: translateX(25px);
    opacity: 1;
  }
  90% {
    transform: translateX(225px);
    opacity: 1;
  }
  100% {
    transform: translateX(250px);
    opacity: 0;
  }
}

input.error {
  border-color: var(--color-third);
}

.input-feedback {
  color: var(--color-third);
  margin-top: var(--size-small);
  font-size: var(--size-small-medium);
}

@media only screen and (max-width: 600px) {
  .pacman {
    display: none;
  }
  .register-container {
    border-radius: 0px;
  }
}

@media only screen and (max-height: 550px) {
  .pacman {
    display: none;
  }
  .container-register {
    width: 100vw;
    height: 100vh;
  }

  .register-container {
    width: 100vw;
    height: 100vh;
  }

  .register-container {
    border-radius: 0px;
  }
}
