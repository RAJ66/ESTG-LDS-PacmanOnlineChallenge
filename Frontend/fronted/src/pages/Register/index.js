import React, { useState } from "react";

import "./styles.css";
import Pacman from "./../../components/Pacman/index";

export default function Register() {
  const [email, setEmail] = useState("");
  const [user, setUser] = useState("");
  const [pass, setPass] = useState("");

  return (
    <div className="container-register">
      <form className="register-container">
        <Pacman className="register-pacman" />
        <input
          placeholder="UserName"
          value={user}
          onChange={event => setUser(event.target.value)}
        />
        <input
          placeholder="Email"
          value={email}
          onChange={event => setEmail(event.target.value)}
        />
        <input
          placeholder="Password"
          type="password"
          value={pass}
          onChange={event => setPass(event.target.value)}
        />

        <button type="submit" className="glow-on-hover">
          Register
        </button>
      </form>
    </div>
  );
}
