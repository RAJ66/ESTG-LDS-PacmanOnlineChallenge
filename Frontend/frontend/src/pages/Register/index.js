import React from "react";

import "./styles.css";
import Pacman from "./../../components/Pacman/index";
import ValidatedRegisterForm from "../../components/ValidatedRegisterForm/index";

export default function Register({ history }) {
  return (
    <div className="container-register">
      <div className="register-container">
        <Pacman className="register-pacman" />
        <ValidatedRegisterForm history={history} />
      </div>
    </div>
  );
}
