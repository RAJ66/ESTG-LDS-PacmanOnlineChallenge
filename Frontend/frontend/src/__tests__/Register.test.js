import React from "react";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";

import Register from "./../pages/Register/index";

describe("Testing Pacman Component", () => {
  it("should render correctly", () => {
    const wrapper = mount(<Register />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
