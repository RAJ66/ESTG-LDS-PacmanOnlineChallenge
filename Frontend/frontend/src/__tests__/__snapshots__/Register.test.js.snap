// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Pacman Component should render correctly 1`] = `
<Register>
  <div
    className="container-register"
  >
    <div
      className="register-container"
    >
      <Pacman
        className="register-pacman"
      >
        <div
          className="pacman"
        >
          <div
            className="pacmanEye"
          />
          <div
            className="pacmanMouth"
          />
          <div
            className="pacmanFood"
          />
        </div>
      </Pacman>
      <ValidatedRegisterForm>
        <Formik
          initialValues={
            Object {
              "email": "",
              "password": "",
              "userName": "",
            }
          }
          onSubmit={[Function]}
          validationSchema={
            ObjectSchema {
              "_blacklist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "_conditions": Array [],
              "_defaultDefault": [Function],
              "_deps": Array [],
              "_excludedEdges": Array [],
              "_exclusive": Object {},
              "_mutate": undefined,
              "_nodes": Array [
                "password",
                "email",
                "userName",
              ],
              "_options": Object {
                "abortEarly": true,
                "recursive": true,
              },
              "_type": "object",
              "_typeError": [Function],
              "_whitelist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "fields": Object {
                "email": StringSchema {
                  "_blacklist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_conditions": Array [],
                  "_deps": Array [],
                  "_exclusive": Object {
                    "required": true,
                    "undefined": false,
                  },
                  "_mutate": undefined,
                  "_options": Object {
                    "abortEarly": true,
                    "recursive": true,
                  },
                  "_type": "string",
                  "_typeError": [Function],
                  "_whitelist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "tests": Array [
                    [Function],
                    [Function],
                  ],
                  "transforms": Array [
                    [Function],
                  ],
                },
                "password": StringSchema {
                  "_blacklist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_conditions": Array [],
                  "_deps": Array [],
                  "_exclusive": Object {
                    "min": true,
                    "required": true,
                    "undefined": false,
                  },
                  "_mutate": undefined,
                  "_options": Object {
                    "abortEarly": true,
                    "recursive": true,
                  },
                  "_type": "string",
                  "_typeError": [Function],
                  "_whitelist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "tests": Array [
                    [Function],
                    [Function],
                    [Function],
                  ],
                  "transforms": Array [
                    [Function],
                  ],
                },
                "userName": StringSchema {
                  "_blacklist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_conditions": Array [],
                  "_deps": Array [],
                  "_exclusive": Object {
                    "min": true,
                    "required": true,
                  },
                  "_mutate": undefined,
                  "_options": Object {
                    "abortEarly": true,
                    "recursive": true,
                  },
                  "_type": "string",
                  "_typeError": [Function],
                  "_whitelist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "tests": Array [
                    [Function],
                    [Function],
                  ],
                  "transforms": Array [
                    [Function],
                  ],
                },
              },
              "tests": Array [],
              "transforms": Array [
                [Function],
              ],
            }
          }
        >
          <form
            className="register-container"
            onSubmit={[Function]}
          >
            <input
              name="userName"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Username"
              type="text"
              value=""
            />
            <input
              name="email"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Email"
              type="text"
              value=""
            />
            <input
              name="password"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />
            <button
              className="glow-on-hover"
              disabled={false}
              type="submit"
            >
              Register
            </button>
          </form>
        </Formik>
      </ValidatedRegisterForm>
    </div>
  </div>
</Register>
`;
